Source/JavaScriptCore/interpreter/Interpreter.cpp

```diff
@@ -451,6 +452,66 @@ bool Interpreter::isOpcode(Opcode opcode)
 }
 #endif // ASSERT_ENABLED
 
+
+void Interpreter::getAsyncStackTrace(JSCell* owner, Vector<StackFrame>& results, JSGenerator* generator, size_t maxStackSize)
+{
+    RELEASE_ASSERT(Options::useAsyncStackTrace());
+    ASSERT(generator);
+
+    AssertNoGC assertNoGC;
+
+    VM& vm = this->vm();
+
+    auto getParentGenerator = [&](JSGenerator* gen) -> JSGenerator* {
+        JSValue contextValue = gen->internalField(static_cast<unsigned>(JSGenerator::Field::Context)).get();
+        JSPromise* awaitedPromise = jsDynamicCast<JSPromise*>(contextValue);
+        if (awaitedPromise && awaitedPromise->status(vm) == JSPromise::Status::Pending) {
+            JSValue reactionsValue = awaitedPromise->internalField(JSPromise::Field::ReactionsOrResult).get();
+            if (JSObject* reactions = jsDynamicCast<JSObject*>(reactionsValue)) {
+                // The @context field is always at offset 3 in Promise reactions objects
+                // created by pushNewPromiseReaction in PromiseOperations.js
+                constexpr PropertyOffset contextPropertyOffset = 3;
+                JSValue contextFieldValue = reactions->getDirect(contextPropertyOffset);
+                if (auto* resultGenerator = jsDynamicCast<JSGenerator*>(contextFieldValue))
+                    return resultGenerator;
+            }
+        }
+        return nullptr;
+    };
+
+    auto computeBytecodeIndex = [&](CodeBlock* codeBlock, JSGenerator* generator) -> BytecodeIndex {
+        BytecodeIndex bytecodeIndex(0);
+        JSValue stateValue = generator->internalField(static_cast<unsigned>(JSGenerator::Field::State)).get();
+        if (stateValue.isInt32()) {
+            int32_t state = stateValue.asInt32();
+            if (state > 0 && codeBlock->numberOfUnlinkedSwitchJumpTables() > 0) {
+                const UnlinkedSimpleJumpTable& jumpTable = codeBlock->unlinkedSwitchJumpTable(0);
```

Comment:
I think this is the last one, correct? Since Generatorification inserts a switch after all bytecode generation gets done. Can you add a test which has integer switch inside generator to test this case?